Source/Plugins/Core/com.equella.core/src/com/tle/core/settings/service/impl/ThemeSettingsServiceImpl.java

```diff
@@ -146,4 +157,63 @@ private String themeToJSONString(NewUITheme theme) throws JsonProcessingExceptio
     themeToString = objectMapper.writeValueAsString(theme);
     return themeToString;
   }
+
+  public String newUIToSass(NewUITheme theme) throws IOException {
+    Map<String, ?> themeVars = objectMapper.readValue(themeToJSONString(theme), Map.class);
+    StringBuilder sassTheme = new StringBuilder();
+
+    themeVars.forEach(
+        (key, value) -> {
+          sassTheme.append("$" + key + " : " + value + ";");
+        });
+
+    return sassTheme.toString();
+  }
+
+  public InputStream getLegacyCss() throws IOException {
+    File baseLegacySass =
+        new File(getClass().getResource("/web/sass/" + SASS_LEGACY_CSS_FILENAME).getFile());
+    File baseLegacyOutput;
+    CustomisationFile customisationFile = new CustomisationFile();
+
+    if (fileSystemService.fileExists(customisationFile, LEGACY_CSS_FILENAME)) {
+      if (baseLegacySass.lastModified()
+          > fileSystemService.lastModified(customisationFile, LEGACY_CSS_FILENAME)) {
+        compileSass();
+      }
+    } else {
+      compileSass();
+    }
+    return fileSystemService.read(customisationFile, LEGACY_CSS_FILENAME);
+  }
+
+  public InputStream compileSass() throws IOException {
```

Comment:
As mentioned in the `interface`, seems this only needs to be private to this class:

```suggestion
  private InputStream compileSass() throws IOException {
```